### DISCUSSION TITLE
Data not retrieved after switching storage backend from Thrift to CQL

### DISCUSSION THREAD

**Discussion by ash74268** [JanusGraph/janusgraph#3067] [Q&A]
Hi All,
I am working on upgrading Cassandra version for my application from 2.1.9 to 3.11.13. We also use JanusGraph 0.3.2 which was using "thrift" as a storage backend. As part of the Cassandra upgrade to 3.11, I have also switched the storage backed to "cql" and update my Java application to use janusgraph-cql driver.  

After upgrading Cassandra and restoring existing data  (including SSTable Upgrade),  the application came up properly but I do not see any data from the JanusGraph which was created using thrift storage backend.  I have verified the data is not lost by using gremlin and querying the old vertices by its names.

Edit:  I have tried creating new vertices and that works. Only issue is that I am not able to query old data from my Java application using janusgraph-cql drivers.

What could possibly be the reason for not retrieving the data previously created with Thrift storage backend?
Any pointers will be appreciated.

**Comment by li-boxuan** [comment]
Not sure of the reason, but I would try and see if I am able to create new vertices to verify the connection between JanusGraph and Cassandra works.

**Reply by ash74268** [reply]
Yes, I have tried creating new vertices and that works. Only issue is that I am not able to query old data from my Java application using janusgraph-cql drivers.. updated my post.

**Reply by li-boxuan** [reply]
And you can use janusgraph-cql driver to read new vertices but not old vertices?

**Reply by ash74268** [reply]
I am using the same janusgraph-cql driver to read old as well as newly created vertices. Only difference is, I dont get any results for old data when using janusgraph-cql.  Shall I my post my janusgraph.properties configuration here?

**Reply by li-boxuan** [reply]
That sounds weird... sure, posting your configuration might help.

**Reply by li-boxuan** [reply]
Also, I don't quite understand what you mean by

> I have verified the data is not lost by using gremlin and querying the old vertices by its names.

How did you verify it using gremlin if your driver cannot read old data?
